storage/database/db_migration.go

```diff
@@ -53,45 +54,56 @@ func (dbm *databaseManager) StartDBMigration(dstdbm DBManager) error {
 
 	// vars for log
 	start := time.Now()
-	iterateNum, fetchedTotal, fetched := 0, 0, 1
+	cycleNum, fetchedTotal, fetchedCycle, fetched := 0, 0, 0, 1
 
 loop:
 	for fetched > 0 {
-		// fetch keys and values
-		var keys, vals [][]byte
-		iterStart := time.Now()
-		keys, vals, fetched = iterateDB(srcIter, dbMigrationFetchNum)
-		elapsedFetchTime := time.Since(iterStart)
-		fetchedTotal += fetched
-
-		// write fetched keys and values to DB
-		putStart := time.Now()
-		for i := 0; i < fetched; i++ {
+		keys := make([][]byte, dbMigrationFetchNum)
+		vals := make([][]byte, dbMigrationFetchNum)
+
+		cycleStart := time.Now()
+		for fetched = 0; fetched < dbMigrationFetchNum && srcIter.Next(); fetched++ {
+			// fetch keys and values
+			// Contents of srcIter.Key() and srcIter.Value() should not be modified, and
+			// only valid until the next call to Next.
+			keys[fetched] = make([]byte, len(srcIter.Key()))
+			vals[fetched] = make([]byte, len(srcIter.Value()))
+			copy(keys[fetched], srcIter.Key())
+			copy(vals[fetched], srcIter.Value())
+
+			// write fetched keys and values to DB
 			// If dstDB is dynamoDB, Put will Write when the number items reach dynamoBatchSize.
-			if err := dstBatch.Put(keys[i], vals[i]); err != nil {
+			if err := dstBatch.Put(keys[fetched], vals[fetched]); err != nil {
 				return errors.Wrap(err, "failed to put batch")
 			}
 		}
-		logger.Info("DB migrated", "iterNum", iterateNum, "fetched", fetched,
-			"elapsedFetch", elapsedFetchTime, "elapsedPut", time.Since(putStart), "elapsedIter", time.Since(iterStart),
-			"fetchedTotal", fetchedTotal, "elapsedTotal", time.Since(start))
+		fetchedTotal += fetched
+		fetchedCycle += fetched
```

Comment:
We can store and use a previous `fetchedTotal` value in the following `if` branch instead of adding every `fetched` value for `fetchedCycle`.

Reply:
review reflected :)

Reply:
thank you!